packages/core/admin/admin/src/content-manager/components/BlocksEditor/index.js

```diff
@@ -43,6 +48,26 @@ const BlocksEditor = React.forwardRef(({ intlLabel, name, readOnly, required, er
       )
     : name;
 
+  const handleToggleMediaLib = () => setMediaLibVisible((prev) => !prev);
+
+  const insertImages = (images) => {
+    images.forEach((img) => {
+      const image = { type: 'image', image: img, children: [{ text: '' }] }; // required empty text node as children for void elements such as Image
+      Transforms.insertNodes(editor, image);
+    });
+  };
+
+  const handleSelectAssets = (images) => {
+    const formattedImages = images.map((image) => ({
+      ...image,
```

Comment:
@markkaylor this adds all image file data from media library, which also has some extra options like isSelected, do you think it's a problem for schema?
Also I have not verified all props yet, will do if needed, please let me know.

Reply:
Hey yes we need to omit the fields not found on the schema otherwise it will throw a validation error. You can see how they are defined here on my PR:

https://github.com/strapi/strapi/pull/18079/files#diff-ca00dda7b24eb88429e1116532b8034076e52cfcfd79da58f8fdcbf9669566efR93-R109

Similarly the response for blocks images should match the response for field images, for example this is the response for field "cover" on address:

![Screenshot 2023-09-19 at 16 13 04](https://github.com/strapi/strapi/assets/26598053/6efa6950-65c9-4500-ad6e-5a2a3c7eeb10)